third_party/terraform/resources/resource_storage_transfer_job.go

```diff
@@ -0,0 +1,529 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/storagetransfer/v1"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+)
+
+func resourceStorageTransferJob() *schema.Resource {
+	return &schema.Resource{
+
+		Create: resourceStorageTransferJobCreate,
+		Read:   resourceStorageTransferJobRead,
+		Update: resourceStorageTransferJobUpdate,
+		Delete: resourceStorageTransferJobDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceStorageTransferJobStateImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
[StringLenBetween](https://github.com/hashicorp/terraform/blob/master/helper/validation/validation.go#L151) can be used here instead I think? Ranging from [0, 1024]